2-Draft-Accepted/RFCNNNN-PSCore6-and-Windows-Modules.md

```diff
@@ -0,0 +1,83 @@
+
+  1. User profile: $env:UserProfile\Documents\WindowsPowerShell\Modules
+  2. Program files: $env:ProgramFiles\WindowsPowerShell\Modules
+  3. System32: $env:Windir\System32\WindowsPowerShell\v1.0\Modules
+
+The proposed change is to add *only* the `System32` path to `PSModulePath` on PowerShell Core 6.
+
+Modules in the user profile should be re-installed under PowerShell Core 6.
+Duplicate modules are something that can be addressed in PowerShellGet and outside the scope of this RFC.
+Modules in `Program Files` that may be compatible with PowerShell Core 6 is addressed below.
+
+Users who want to search the entirety of the Windows PowerShell `PSModulePath` should use `Add-WindowsPSModulePath` cmdlet
+from the `WindowsCompatibility` module.
+
+### PowerShell Core PSModulePath
+
+Modules in the PowerShell Core `PSModulePath` are implicitly deemed compatible even if the manifest does not
+declare compatibility.
+A user who explicitly uses `Install-Module` within PowerShell Core 6 expect those modules to be available to PowerShell Core 6.
+
+### CompatiblePSEditions for Windows System32
+
+When we add the `System32` path to PowerShell Core's `PSModulePath`,
+we will use additional logic to validate that the module is compatible with PowerShell Core.
+
+If the module manifest contains `CompatiblePSEditions` with value `Core`, then that module is treated as compatible.
+If the module manifest contains `CompatiblePSEditions` with only the value of `Desktop`, then that module is treated as incompatible.
+If the module manifest does not contain `CompatiblePSEditions`, then that module is treated the same as `Desktop` which is incompatible.
+
+By default, `Get-Module -ListAvailable` will only show modules found in `PSModulePath` that are declared as compatible.
+Similarly, module auto-discovery and auto-loading will only find modules that are declared as compatible.
+
+The `ModuleInfo` table format will add a column for `CompatiblePSEditions` (between `Name` and `ExportedCommands`).
+
+### Using Modules not declared as Compatible
+
+`Get-Module -ListAvailable` will support `-SkipCompatibilityCheck` switch to list all modules in `PSModulePath`.
+
+If `Import-Module` is used with a module that is not declared in the manifest as compatible, it will throw a terminating error:
+
+> The module '{0}' could not be imported as it is not declared as compatible.  Use `-SkipCompatibilityCheck` switch to override this check.
+
+`Import-Module` will support `-SkipCompatibilityCheck` switch to try and import the module regardless of `CompatiblePSEditions`.
+Since module auto-discovery does not expose cmdlets from modules not declared as compatible, explicit `Import-Module` with
+`-SkipCompatibilityCheck` is required to use potentially incompatible modules.
+
+## Alternate Proposals and Considerations
+
+Open issue: Should `CompatiblePSEditions` column only be added in `ModuleInfo` table format if used with `-SkipCompatibilityCheck`?
```

Comment:
I believe we should show this always - we should inform users what modules is still not compatible to exclude hidden errors.

Reply:
We can see what the user feedback is once @rjmholt has some working code checked in.  My only concern is taking up too much column space.  Although in table formatting, we could call it `PSEdition` and perhaps just `D` or `C` instead of `Desktop` and `Core` if we need to save space.